### PR TITLE
Add stopping criteria

### PR DESCRIPTION
**Pull request by benhsu75**
Asana Task: https://app.asana.com/0/38684300131201/357028383074880

One potentially controversial change I made is added the LinkedResource class. I commented what this does, but the reason I did this was because StoppingCriteria is actually the first resource that doesn't have its own id/wouldn't be POSTed or UPDATEd and isn't a paginated list (like best_assignments). Since I could not add stopping criteria to the client with the Subresource.fetch() function, I thought it made sense that while something like observation was a "subresource,"  stopping_criteria was more of a "linked resource."

Let me know if there is another way of doing this that makes more sense.

If we did it this way, I'd also update the documentation which says that the developer accesses stopping_criteria with 

`new Experiment(EXPERIMENT_ID).stopping_criteria().list().call()`

^That didn't make sense to me since the response to /stopping_criteria is not a paginated list.

### CODE REVIEW

File: src/main/java/com/sigopt/model/Experiment.java [hunk 1]
```diff
@@ -118,6 +118,26 @@ public Integer getFolds() {
         return Experiment.delete().addPathComponent("id", id);
     }
 
+    /**
+    *   LinkedResource is used for a resource linked strongly to another e.g.
```
Comment (alexandraj777) [inline src/main/java/com/sigopt/model/Experiment.java]: I like your thinking! I would think something like PropertyResource, since stopping criteria is a property while observations are a subresource. This is also because `linked` has such a strong connotation for programmers already and is tied to object structure, lists, etc.

File: src/main/java/com/sigopt/model/Experiment.java [hunk 2]
```diff
@@ -153,7 +153,7 @@ public Subresource(String prefix, String name, Class<T> klass) {
         }
 
         public APIMethodCaller<T> fetch(String id) {
-            return this.fetch().addParam("id", id);
+            return this.fetch().addPathComponent("id", id);
```
Comment (alexandraj777) [inline src/main/java/com/sigopt/model/Experiment.java]: Is this for the suggestion fix? If so, could it be a separate PR? Thanks :)

Comment (benhsu75) [inline src/main/java/com/sigopt/model/Experiment.java]: https://sigopt.com/docs/endpoints/suggestions/detail and https://sigopt.com/docs/endpoints/observations/detail were actually broken before b/c of this line. Working now.

**Comment by benhsu75** [comment]
@alexandraj777 Changed naming and fixed a one line bug. Lmk if it looks good to go.

File: src/main/java/com/sigopt/model/Experiment.java (hunk 2 above)
Comment (benhsu75) [inline src/main/java/com/sigopt/model/Experiment.java]: yep, can do
